core/keystore/src/lib.rs

```diff
@@ -151,22 +145,126 @@ impl Store {
 		Ok(public_keys)
 	}
 
-	fn key_file_path<TPair: Pair + TypedKey>(&self, public: &TPair::Public) -> PathBuf {
+	fn key_file_path<TPair: Pair>(&self, public: &TPair::Public) -> PathBuf {
 		let mut buf = self.path.clone();
 		let bytes: [u8; 4] = TPair::KEY_TYPE.to_le_bytes();
 		let key_type = hex::encode(bytes);
 		let key = hex::encode(public.as_slice());
 		buf.push(key_type + key.as_str());
 		buf
 	}
+
+	/// Register a `StoreHandler`.
+	fn register_handler<TKey: TypedKey, THandler: StoreHandler + 'static>(
```

Comment:
Can this be `pub`? It could be useful to add `NetworkService` here as a handler so changes to session keys can get propagated to the network DHT right away.

Reply:
I'd rather not pass byte vecs around, this is private to avoid that.